src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Integer.Signed.N.cs

```diff
@@ -8,12 +8,186 @@ public static partial class Utf8Parser
     {
         private static bool TryParseSByteN(ReadOnlySpan<byte> text, out sbyte value, out int bytesConsumed)
         {
-            throw NotImplemented.ActiveIssue("https://github.com/dotnet/corefx/issues/24986");
+            if (text.Length < 1)
+                goto FalseExit;
+
+            int sign = 1;
+            int index = 0;
+            int c = text[index];
+            if (c == '-')
```

Comment:
This exact same code is duplicated a few times throughout the new methods in the PR.  Can this be consolidated?
```
            if (c == '-')
             {
                 sign = -1;
                 index++;
                 if ((uint)index >= (uint)text.Length)
                     goto FalseExit;
                 c = text[index];
             }
             else if (c == '+')
             {
                 index++;
                 if ((uint)index >= (uint)text.Length)
                     goto FalseExit;
                 c = text[index];
             }
```